### PR TITLE
Change queue of the sending thread

### PR DESCRIPTION
**Pull request by gichan-jang**
- Release send queue after the send thread is finished.
 - Set timeout for send queue to terminate the send thread.

Signed-off-by: gichan <gichan2.jang@samsung.com>

### CODE REVIEW

File: src/libnnstreamer-edge/nnstreamer-edge-internal.c [hunk 1]
```diff
@@ -812,8 +812,13 @@ _nns_edge_send_thread (void *thread_data)
   int64_t client_id;
   char *val;
   int ret;
+  uint queue_timeout_sec = 3000U;
 
-  while (nns_edge_queue_wait_pop (eh->send_queue, 0U, &data_h, &data_size)) {
+  if (NNS_EDGE_NODE_TYPE_PUB == eh->node_type)
+    queue_timeout_sec = 0U;
+
+  while (nns_edge_queue_wait_pop (eh->send_queue, queue_timeout_sec,
```
Comment (jaeyun-jung) [inline src/libnnstreamer-edge/nnstreamer-edge-internal.c:820]: Adding timeout here may cause an error if edge node sends new data after 3 seconds.
I suggest adding new function nns_edge_queue_stop () to stop and clear data in the queue.

File: src/libnnstreamer-edge/nnstreamer-edge-internal.c [hunk 2]
```diff
@@ -1378,13 +1383,12 @@ nns_edge_release_handle (nns_edge_h edge_h)
   eh->user_data = NULL;
   eh->broker_h = NULL;
 
-  nns_edge_queue_destroy (eh->send_queue);
```
Comment (jaeyun-jung) [inline src/libnnstreamer-edge/nnstreamer-edge-internal.c:1381]:
```
/**
 * @brief Stop waiting for new data and clear all data in the queue.
 */
void
nns_edge_queue_stop (gueue_handle handle, bool clear)
{
  nns_edge_queue_s *q = (nns_edge_queue_s *) handle;

  if (!q) {
    nns_edge_loge ("[Queue] Invalid param, queue is null.");
    return false;
  }

  nns_edge_lock (q);
  nns_edge_cond_signal (q);

  if (clear) {
    while (q->length > 0U)
      _pop_data (q, true, NULL, NULL);
  }

  nns_edge_unlock (q);
}
```

need flag to stop sending thread here.
```
nns_edge_queue_stop (eh->send_queue, true);
  if (eh->send_thread) {
    eh->sending = false;
    pthread_join (eh->send_thread, NULL);
    eh->send_thread = 0;
  }
```

Comment (gichan-jang) [reply src/libnnstreamer-edge/nnstreamer-edge-internal.c:1381]: Thanks for the suggestion! Updated.

File: src/libnnstreamer-edge/nnstreamer-edge-queue.h [hunk 3]
```diff
@@ -92,6 +92,15 @@ bool nns_edge_queue_pop (nns_edge_queue_h handle, void **data, nns_size_t *size)
  */
 bool nns_edge_queue_wait_pop (nns_edge_queue_h handle, unsigned int timeout, void **data, nns_size_t *size);
 
+/**
+ * @brief Stop waiting for new data and clear all data in the queue.
+ * @param[in] handle The queue handle.
+ * @return true on success.
+ * @note When this function is called, nns_edge_queue_wait_pop will stop the waiting.
+ */
+bool
```
Comment (jaeyun-jung) [inline src/libnnstreamer-edge/nnstreamer-edge-queue.h:101]: plz fix code style, tizen api declaration.
`bool nns_edge_queue_clear (nns_edge_queue_h handle);`

**Pull request merged**